### PR TITLE
Introduce glossary

### PR DESCRIPTION
**Pull request by msehnout**
It is by far not exhaustive, but it could serve as a starting point.

### CODE REVIEW

File: osbuild-composer/src/developer-guide/glossary.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+| Term                 | Explanation                                                  |
+| -------------------- | ------------------------------------------------------------ |
+| AMI                  | Amazon Machine Image (image type)                            |
+| Blueprint            | Definition of customizations in the image                    |
+| Compose              | Request from the user. osbuild-composer running the Weldr API can only create one image at a time, so one compose maps directly to one image build. |
+| Composer API         | HTTP API meant as publicly accessible (over TCP). It was created specifically for osbuild-composer and does not support some Weldr features like blueprint management, but adds new features like building different distros and architectures. |
+| Image Build          | One request from osbuild-composer to osbuild-worker. Its result is a single image. |
+| Image Type           | Image file format usually associated with a specific use case. For example: AMI for AWS, qcow2 for OpenStack, etc. |
+| Manifest             | Input for the osbuild tool. It should be a precise definition of an image. |
```
Comment (achilleas-k) [inline osbuild-composer/src/developer-guide/glossary.md:11]: Could link to https://www.osbuild.org/man/osbuild-manifest.5 for more information and for the reader to get an idea of what a manifest looks like (what kind of information it contains).

File: osbuild-composer/src/developer-guide/glossary.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Glossary
+
+| Term                 | Explanation                                                  |
+| -------------------- | ------------------------------------------------------------ |
+| AMI                  | Amazon Machine Image (image type)                            |
+| Blueprint            | Definition of customizations in the image                    |
+| Compose              | Request from the user. osbuild-composer running the Weldr API can only create one image at a time, so one compose maps directly to one image build. |
```
Comment (achilleas-k) [inline osbuild-composer/src/developer-guide/glossary.md:7]: Should we also mention that it _can_ map to multiple images (in the case of Koji)?

**Review by achilleas-k — APPROVED** [review]
This is great!  Thanks for writing this.  I have some minor recommendations but you can ignore them if you think they don't fit (or don't make sense).

File: osbuild-composer/src/developer-guide/glossary.md (hunk 1 above)
Comment (msehnout) [reply osbuild-composer/src/developer-guide/glossary.md:11]: Yes

File: osbuild-composer/src/developer-guide/glossary.md (hunk 2 above)
Comment (msehnout) [reply osbuild-composer/src/developer-guide/glossary.md:7]: Also yes :-)

**Comment by msehnout** [comment]
I've updated to document according to your suggestions. I'll merge as is for now and we can continue in follow up PRs.

**Pull request merged**